runtime/lua/vim/filetype.lua

```diff
@@ -336,6 +336,7 @@ local extension = {
   cyn = 'cynpp',
   cypher = 'cypher',
   dart = 'dart',
+  dfy = 'dafny',
```

Comment:
This is is sorted by filetype, not extension
```suggestion
  dfy = 'dafny',
  dart = 'dart',
```